packages/replay/src/uploadWorker.ts

```diff
@@ -0,0 +1,70 @@
+import type { URL } from "url";
+import type { AgentOptions } from "http";
+
+import { parentPort } from "worker_threads";
+import fs from "fs";
+import fetch from "node-fetch";
+import { Agent as HttpAgent } from "http";
+import { Agent as HttpsAgent } from "https";
+import { getUserAgent } from "./utils";
+import dbg from "./debug";
+
+const agentConfig: AgentOptions = {
+  keepAlive: true,
+  maxSockets: 500,
+};
+const agent = (parsedURL: URL) =>
+  parsedURL.protocol == "http:" ? new HttpAgent(agentConfig) : new HttpsAgent(agentConfig);
+
+if (parentPort === null) {
+  throw new Error("Must be run as a worker");
+}
+
+parentPort.on(
+  "message",
+  async ({
+    link,
+    partMeta,
+    size,
+    logPath,
+  }: {
+    link: string;
+    partMeta: { filePath: string; start: number; end: number };
+    size: number;
+    logPath: string;
+  }) => {
+    const { filePath, start, end } = partMeta;
+    const debug = dbg("replay:cli:upload-worker", logPath);
+
+    if (parentPort === null) {
+      throw new Error("Must be run as a worker");
+    }
+
+    debug(`Uploading chunk size: ${size}`);
```

Comment:
I'd include the part metadata here too. It's likely that these logs will be interleaved so including more data elements in the logs will help us correlate.